framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -10040,7 +10033,7 @@ void VulkanReplayConsumerBase::OverrideDestroyPipeline(
 
     if (pipeline_info != nullptr)
     {
-        VkPipeline in_pipeline =
+        in_pipeline =
```

Comment:
this is the main issue here. instead of assigning to `in_pipeline`, a local variable with same name was created.

Reply:
the symptom can be seen using `--validate`, we currently leak `VkPipeline` objects, spilling validation-warnings when the capture terminates: ```vkDestroyDevice():  OBJ ERROR : For VkDevice 0x5cf6ebc66d00[], VkPipeline 0x421b27000000016d[] has not been destroyed.``` - this is fixed here